src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+
+import { clsx } from 'clsx';
+import React, { useEffect } from 'react';
+
+import Button from '../Button/Button';
+
+export type ToastVariant = 'success' | 'error' | 'info' | 'warning';
+
+interface ToastProps {
+  id: string;
+  message: React.ReactNode;
+  variant?: ToastVariant;
+  duration?: number;
+  onClose: (id: string) => void;
+}
+
+export default function Toast({
+  id,
+  message,
+  variant = 'info',
+  duration = 3000,
+  onClose,
+}: ToastProps) {
+  // 자동 닫기
+  useEffect(() => {
+    if (duration > 0) {
+      const t = setTimeout(() => onClose(id), duration);
+      return () => clearTimeout(t);
+    }
+  }, [duration, id, onClose]);
+
+  // Toast 배경 색상 매핑
+  const variantStyles: Record<ToastVariant, string> = {
+    success: 'bg-green-500 text-white',
+    error: 'bg-red-500 text-white',
+    info: 'bg-blue-500 text-white',
+    warning: 'bg-yellow-500 text-black',
+  };
```

Comment:
현재 토스트 알림마다 새로운 객체를 생성하는 구조로,
Tailwind 클래스 상수를 외부에 정의하여 재사용할 수 있습니다.
```suggestion
const VARIANT_STYLES: Record<ToastVariant, string> = {
    success: 'bg-green-500 text-white',
    error: 'bg-red-500 text-white',
    info: 'bg-blue-500 text-white',
    warning: 'bg-yellow-500 text-black',
};

interface ToastProps {
    ...
```